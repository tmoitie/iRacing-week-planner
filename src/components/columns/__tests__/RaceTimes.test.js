import moment from 'moment';
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, test, expect } from '@jest/globals';
import { TableWrapper } from './ColumnUtils';

import RaceTimes from '../RaceTimes';

describe('components/columns/RaceTimes', () => {
  test('renders no time data', () => {
    render(
      <TableWrapper>
        <RaceTimes
          race={{
            raceTimes: null,
          }}
        />
      </TableWrapper>,
    );

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
  });

  test('renders settimes', async () => {
    const user = userEvent.setup();
    const { baseElement } = render(
      <TableWrapper>
        <RaceTimes
          race={{
            series: 'Mazda Cup',
            setTimes: [
              moment.duration({ days: 3, hours: 2 }),
              moment.duration({ days: 4, hours: 13 }),
            ],
          }}
        />
      </TableWrapper>,
    );

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();

    await user.click(screen.getByText('Set times'));
    expect(baseElement).toMatchSnapshot();

    await user.click(screen.getByText('Close'));
    expect(baseElement).toMatchSnapshot();
  });

  test('renders everytime', () => {
    render(
      <TableWrapper>
        <RaceTimes
          race={{
            series: 'Mazda Cup',
            everyTime: moment.duration({ hours: 3 }),
            offset: moment.duration({ minutes: 90 }),
          }}
        />
      </TableWrapper>,
    );

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
  });

  test('renders everytime no offset', () => {
    render(
      <TableWrapper>
        <RaceTimes
          race={{
            series: 'Mazda Cup',
            everyTime: moment.duration({ hours: 3 }),
          }}
        />
      </TableWrapper>,
    );

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
  });
});
