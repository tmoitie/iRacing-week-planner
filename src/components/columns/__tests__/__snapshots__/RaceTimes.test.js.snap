// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/RaceTimes renders everytime 1`] = `
<td>
  <div>
    Every 3 hours starting at 1:30 AM UTC
  </div>
</td>
`;

exports[`components/columns/RaceTimes renders everytime no offset 1`] = `
<td>
  <div>
    Every 3 hours starting at 12:00 AM UTC
  </div>
</td>
`;

exports[`components/columns/RaceTimes renders no time data 1`] = `
<td>
  No time data
</td>
`;

exports[`components/columns/RaceTimes renders settimes 1`] = `
<td
  class="clickableCell"
>
  <button
    class="cellButton"
    style="min-height: 0;"
    type="button"
  >
    Set times
  </button>
</td>
`;

exports[`components/columns/RaceTimes renders settimes 2`] = `
<body>
  <div>
    <table>
      <tbody>
        <tr
          data-testid="table-row"
        >
          <td
            class="clickableCell"
          >
            <button
              class="cellButton"
              style="min-height: 0;"
              type="button"
            >
              Set times
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    id="undefined"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; display: block; inset: 0; background-color: rgba(255, 255, 255, 0.75);"
    >
      <div
        class="Modal__Bootstrap modal-dialog modal-lg"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <button
              aria-label="Close"
              class="close"
              type="button"
            >
              <span
                class="glyphicon glyphicon-remove"
              />
            </button>
            <h4
              class="modal-title"
            >
              Set times for Mazda Cup
            </h4>
          </div>
          <div
            class="modal-body"
            style="max-height: 55vh; overflow-y: auto;"
          >
            <div
              class="container-fluid"
            >
              <ul>
                <li>
                  Friday, 2:00 AM GMT (Friday, 2:00 AM UTC)
                </li>
                <li>
                  Saturday, 1:00 PM GMT (Saturday, 1:00 PM UTC)
                </li>
              </ul>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`components/columns/RaceTimes renders settimes 3`] = `
<body>
  <div>
    <table>
      <tbody>
        <tr
          data-testid="table-row"
        >
          <td
            class="clickableCell"
          >
            <button
              class="cellButton"
              style="min-height: 0;"
              type="button"
            >
              Set times
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    id="undefined"
  />
</body>
`;
